test/ZeroState.t.sol

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import "forge-std/Test.sol";
+import "src/Rings.sol";
+import "src/WRings.sol";
+
+abstract contract ZeroState is Test {
+    
+    Rings public rings;
+    WRings public wrings;
+
+    address sonic = address(0x1);
+    address tails = address(0x2);
+    address knuckles = address(0x3);
+
+    address[] users = [sonic, tails, knuckles];
+    
+    function setUp() public virtual {
+        rings = new Rings();
+        wrings = new WRings(rings);
+
+        for(uint i = 0; i < users.length; i++) {
+            rings.mint(users[i], 1000*10**18);
```

Comment:
nicer to have spaces between most math operators unless u r intentionally grouping things.  in this case i think `1000 * 10**18` is best.  or actually we usually use `1000 * 1e18` 

anyways, just a nit, no need to change

> Operators with a higher priority than others can exclude surrounding whitespace in order to denote precedence. This is meant to allow for improved readability for complex statements. You should always use the same amount of whitespace on either side of an operator:
https://docs.soliditylang.org/en/v0.8.13/style-guide.html#other-recommendations